tests/test_wasm_builtin_simd.c

```diff
@@ -0,0 +1,488 @@
+#include <assert.h>
+#include <stdint.h>
+#include <stdio.h>
+#include <emscripten.h>
+
+typedef char i8x16 __attribute((vector_size(16)));
+typedef short i16x8 __attribute((vector_size(16)));
+typedef int i32x4 __attribute((vector_size(16)));
+typedef long long i64x2 __attribute((vector_size(16)));
+typedef unsigned char u8x16 __attribute((vector_size(16)));
+typedef unsigned short u16x8 __attribute((vector_size(16)));
+typedef unsigned int u32x4 __attribute((vector_size(16)));
+typedef unsigned long long u64x2 __attribute((vector_size(16)));
+typedef float f32x4 __attribute((vector_size(16)));
+typedef double f64x2 __attribute((vector_size(16)));
+typedef u8x16 v128;
+
+#define TESTFN EMSCRIPTEN_KEEPALIVE __attribute__((noinline))
+
+v128 TESTFN v128_load(v128 *ptr) {
+  return *ptr;
+}
+void TESTFN v128_store(v128 *ptr, v128 vec) {
+  *ptr = vec;
+}
+i32x4 TESTFN i32x4_const(void) {
+  return (i32x4) {1, 2, 3, 4};
+}
+// v128 TESTFN v128_shuffle_interleave_bytes(v128 x, v128 y) {
+//   return __builtin_shufflevector(x, y, 0, 17, 2, 19, 4, 21, 6, 23, 8, 25, 10, 27, 12, 29, 14, 31);
+// }
+// i32x4 TESTFN v128_shuffle_reverse_i32s(i32x4 vec) {
+//   return __builtin_shufflevector(vec, vec, 3, 2, 1, 0);
+// }
```

Comment:
Shuffles seem to crash d8.

Reply:
I'll take a look to see what's happening.